### PR TITLE
FT retry checkMetricsForExecutorService

### PR DESCRIPTION
**Pull request by danielkec**
Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java [hunk 1]
```diff
@@ -204,25 +212,16 @@ void checkMetricsForExecutorService() throws Exception {
...
+        retry.invoke(() -> metricsRequestBuilder
+                        .submit()
+                        .peek(res -> assertThat("Second access to metrics", res.status().code(), is(200)))
+                        .flatMapSingle(res -> res.content().as(JsonObject.class))
+                        .peek(json -> assertThat("JSON metrics results after accessing slow endpoint",
+                                json, hasKey(jsonKeyForCompleteTaskCountInThreadPool)))
+                        .map(json -> json.getInt(jsonKeyForCompleteTaskCountInThreadPool))
+                        .forSingle(secCompletedTaskCnt ->
+                                assertThat("Completed task count after accessing slow endpoint", secCompletedTaskCnt, is(1))))
+                .await(CLIENT_TIMEOUT);
```
Comment (tjquinno) [inline metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java:224]: It is appealing to reuse the existing FT impl vs. a retry loop.

The JavaDoc for `peek` does say it's intended mainly for debugging and warns that the stream implementation might optimize away calls to `peek` for some elements of the stream. That probably cannot happen here (there are no `findFirst` or other optimizable steps in handling this stream).

File: metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java [hunk 2]
```diff
@@ -64,7 +65,7 @@ public class TestServer {
     private static MetricsSupport metricsSupport;
 
     private static final Duration CLIENT_TIMEOUT = Duration.ofSeconds(5);
```
Comment (tjquinno) [inline metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java:67]: I had included property look-up of `RETRY_ COUNT` and `RETRY_DELAY_MS` so we would have an easy way to adjust for slow pipelines or action runners.

I know `CLIENT_TIMEOUT` was not originally set up that way, but do you think it is worth doing a similar property look-up for it for the same reason as part of this PR?

**Review by tjquinno — DISMISSED** [review]
Looks good. See comments for a suggestion you might want to include.

File: metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java (hunk 1 above)
Comment (danielkec) [reply metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java:224]: This is our reactive peek, we don't omit side-effects like Java streams does

File: metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java (hunk 2 above)
Comment (danielkec) [reply metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java]: Right

**Review by tjquinno — APPROVED** [review]
LGTM

**Pull request merged**